libclamav/regex/engine.c

```diff
@@ -223,7 +223,7 @@ matcher(struct re_guts *g, const char *string, size_t nmatch,
 			dp = dissect(m, m->coldp, endp, gf, gl);
 		} else {
 			if (g->nplus > 0 && m->lastpos == NULL)
-				m->lastpos = (char **)cli_max_malloc((g->nplus+1) *
+				m->lastpos = (const char **)cli_max_malloc((g->nplus+1) *
```

Comment:
It feels like the correct fix is to remove `const` in the `lastpos` definition.  Check out the warnings in the "Files Changed" tab.